src/spd3303x.rs

```diff
@@ -93,37 +162,27 @@ impl Spd3303x {
         )
     }
 
-    async fn send_raw<Request>(&mut self, request: Request) -> Result<()>
+    async fn send<Request>(&mut self, request: Request) -> Result<()>
     where
-        Request: ScpiRequest,
+        Request: ScpiRequest<Response = EmptyResponse>,
     {
         let mut out = String::with_capacity(128);
         request.serialize(&mut out);
         out.push('\n');
-        self.writer.write_all(out.as_bytes()).await?;
-
-        Ok(())
+        self.driver.send(out.as_str()).await
     }
 
-    async fn send<Request>(&mut self, request: Request) -> Result<()>
-    where
-        Request: ScpiRequest<Response = EmptyResponse>,
-    {
-        self.send_raw(request).await
-    }
     async fn execute<Request, Response>(&mut self, request: Request) -> Result<Response>
     where
         Request: ScpiRequest<Response = Response>,
         Response: ScpiDeserialize,
     {
-        self.send_raw(request).await?;
-
-        let mut line = String::new();
-
-        self.reader.read_line(&mut line).await?;
-        let data = line.as_str();
+        // TODO copy pasted
+        let mut out = String::with_capacity(128);
+        request.serialize(&mut out);
```

Comment:
The fact that `out.push('\n');` is present in `send` serialization but not in `execute` looks very suspicious.
Double check if this is correct, and if yes write a comment.